qcengine/config.py

```diff
@@ -72,7 +74,33 @@ class NodeDescriptor(pydantic.BaseModel):
     jobs_per_node: int = 2
     retries: int = 0
 
-    def __init__(self, **data: Dict[str, Any]) -> "BaseModel":
+    # Cluster options
+    is_batch_node: bool = pydantic.Field(False, help="""Whether the node running QCEngine is a batch node
+    
+    Some clusters are configured such that tasks are launched from a special "batch" or "MOM" onto the compute nodes.
+    The compute nodes on such clusters often have a different CPU architecture than the batch nodes and 
+    often are unable to launch MPI tasks, which has two implications:
+        1) QCEngine must make *all* calls to an executable via ``mpirun`` because the executables might not
+        be able to run on the batch node. 
+        2) QCEngine must run on the batch node to be able to launch tasks on the more than one compute nodes  
+    
+    ``is_batch_node`` is used when creating the task configuration as a means of determining whether
+    ``mpirun_command`` must always be used even for serial jobs (e.g., getting the version number)
+    """)
+    mpirun_command: Optional[str] = pydantic.Field(
+        None, description="""Invocation for launching node-parallel tasks with MPI
+        
+        The invocation need not specify the number of nodes, tasks, or cores per node.
+        Information about the task configuration will be added to the command by use of
+        Python's string formatting. The configuration will be supplied as the following variables:
+        
+            {nnodes} - Number of nodes
+            {ranks_per_node} - Number of MPI ranks per node
+            {total_ranks} - Total number of MPI ranks
```

Comment:
An example here would be great, I didn't see at first how of this slotted together.

Reply:
I also wonder if we need a validator that checks for each of parameters in the string so that we can validate it.

Reply:
Sure. I'll put in an example and a validator is a good idea. However, some MPI launch systems will not use all 3 of these (e.g., the `aprun` used by Cray does not use `nnodes`) so I might make sure either `nnodes` or `total_ranks` is used along with `total_ranks`.

Reply:
```
>>> re.findall("{\w+}", "hello {this} is a {scan} of {paren} words")
['{this}', '{scan}', '{paren}']
```
All together `set(re.findall("{\w+}", ...)) < {"nodes", "ranks", ...}`?

Reply:
@mattwelborn @loriab I see now that everything in life can be solved via `re`.

Reply:
Nice! That's some good regex magic! I I opted for writing a more explicit test using `not in`, but going to have to remember this one :)